probe_scraper/runner.py

```diff
@@ -213,6 +214,47 @@ def dedupe_probes(results):
     return probes
 
 
+def apply_ping_specific_metadata(metadata, ping_metadata):
+    """apply_ping_specific_metadata recurses down into dicts nested
+    to an arbitrary depth, updating keys. The ``ping_metadata`` is merged into
+    ``metadata``.
+    :param metadata: dict onto which the merge is executed
+    :param ping_metadata: dct merged into metadata
+    :return: None
+    """
+    for k, v in ping_metadata.items():
+        if (
+            k in metadata
+            and isinstance(metadata[k], dict)
+            and isinstance(ping_metadata[k], dict)
+        ):
+            apply_ping_specific_metadata(metadata[k], ping_metadata[k])
+        else:
+            metadata[k] = ping_metadata[k]
+
+
+def add_pipeline_metadata(pings_by_repo, repositories):
+    for repo in repositories:
+        metadata_defaults = repo.moz_pipeline_metadata_defaults
+        metadata_defaults["bq_dataset_family"] = repo.app_id.replace("-", "_")
+        metadata_defaults["bq_metadata_format"] = (
+            "pioneer" if repo.app_id.startswith("rally") else "structured"
+        )
+
+        current_pings = pings_by_repo.get(repo.name)
+        for ping_name, ping in current_pings.items():
+            ping_metadata = repo.moz_pipeline_metadata.get(ping_name, {})
+            pipeline_metadata = copy.deepcopy(metadata_defaults)
+            pipeline_metadata["bq_table"] = ping_name.replace("-", "_") + "_v1"
+
+            apply_ping_specific_metadata(pipeline_metadata, ping_metadata)
+
+            if pipeline_metadata:
+                pings_by_repo.get(repo.name).get(ping_name)[
```

Comment:
```suggestion
                ping[
```

I believe `ping` from the `for` statement should still be in scope here and refers to the relevant item within the overall `pings_by_repo` structure.

Reply:
Agreed